### ISSUE TITLE
Node aborted with segmentation fault after 2 consecutive reboots

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylla#9959]
*Installation details*
Scylla version (or git commit hash):4.7.dev-0.20220118.985403ab9 with build-id d48bc5d3676f53140f58fae39086e98f17a58534
Cluster size:4
OS (RHEL/CentOS/Ubuntu/AWS AMI):ami-057906d0f869ccd53 (eu-west-1)
Scenario: https://github.com/scylladb/scylla-cluster-tests/blob/master/test-cases/longevity/longevity-lwt-basic-3h.yaml
### That scenario uses LWT

At `2022-01-23 06:01:26`, a disrupt_multiple_hard_reboot_node nemesis has started on node 3 (54.77.138.191 | 10.0.1.193). As the name implies, the nemesis reboots the target node's instance multiple times.
After the node was rebooted for the second time, at `2022-01-23 06:18:52,347`, the node was aborted by a Segmentation fault:
```
Segmentation fault on shard 0.
Backtrace:
  0x424fef8
  0x4280796
  0x48cebd1
  0x7f5e900d5a1f
  0x1713852
  0x170e5a1
  0x16b7f50
  0x373e1b9
  0x4260e54
  0x4262227
  0x426148c
  0x4207298
  0x4206776
  0xfdae3d
  0xfd871a
  /opt/scylladb/libreloc/libc.so.6+0x27b74
  0xfd76ed
```

Coredump:
```
backtrace=           PID: 797 (scylla)
           UID: 113 (scylla)
           GID: 119 (scylla)
        Signal: 11 (SEGV)
     Timestamp: Sun 2022-01-23 06:22:44 UTC (2min 58s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 100 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1 --enable-sstable-key-validation 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 0-7 --lock-memory=1
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: 4895e16e6b2c4bc9b028baef94132c19
    Machine ID: 122c2e3e25124b2dbcf5ddb2c894c926
      Hostname: longevity-lwt-3h-master-db-node-beaf65ba-3
       Storage: /var/lib/systemd/coredump/core.scylla.113.4895e16e6b2c4bc9b028baef94132c19.797.1642918964000000000000
       Message: Process 797 (scylla) of user 113 dumped core.
                
                Stack trace of thread 797:
                #0  0x0000000001713853 _ZN7seastar8futurizeINS_6futureIvEEE6invokeIRZN12_GLOBAL__N_112shard_reader5closeEvE4$_32JEEES2_OT_DpOT0_ (scylla + 0x1513853)
                #1  0x000000000170e5a2 _ZN12_GLOBAL__N_112shard_reader5closeEv (scylla + 0x150e5a2)
                #2  0x00000000016b7f51 _ZN27multishard_combining_reader5closeEv (scylla + 0x14b7f51)
                #3  0x000000000373e1ba _Z9read_pageIN12_GLOBAL__N_125data_query_result_builderEEN7seastar6futureINS0_19page_consume_resultIT_EEEENS2_10shared_ptrI12read_contextEENS2_13lw_shared_ptrIK6schemaEERKN5query12read_commandERKSt6vectorI20nonwrapping_intervalIN3dht13ring_positionEESaISN_EEN7tracing15trace_state_ptrEOS5_.resume (scylla + 0x353e1ba)
                #4  0x0000000004260e55 _ZN7seastar7reactor14run_some_tasksEv (scylla + 0x4060e55)
                #5  0x0000000004262228 _ZN7seastar7reactor6do_runEv (scylla + 0x4062228)
                #6  0x000000000426148d _ZN7seastar7reactor3runEv (scylla + 0x406148d)
                #7  0x0000000004207299 _ZN7seastar12app_template14run_deprecatedEiPPcOSt8functionIFvvEE (scylla + 0x4007299)
                #8  0x0000000004206777 _ZN7seastar12app_template3runEiPPcOSt8functionIFNS_6futureIiEEvEE (scylla + 0x4006777)
                #9  0x0000000000fdae3e _ZL11scylla_mainiPPc (scylla + 0xddae3e)
                #10 0x0000000000fd871b main (scylla + 0xdd871b)
                #11 0x00007f5e8f504b75 __libc_start_main (libc.so.6 + 0x27b75)
                #12 0x0000000000fd76ee _start (scylla + 0xdd76ee)
...
download_instructions=gsutil cp gs://upload.scylladb.com/core.scylla.113.4895e16e6b2c4bc9b028baef94132c19.797.1642918964000000000000/core.scylla.113.4895e16e6b2c4bc9b028baef94132c19.797.1642918964000000000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.113.4895e16e6b2c4bc9b028baef94132c19.797.1642918964000000000000.gz
```

Logs:
```
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                        Log links for testrun with test id beaf65ba-b35f-4721-bc69-9b17ec0799aa                                                                                                        |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type       | Link                                                                                                                                                                                                                                               |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20220123_080431 | grafana        | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_080431/grafana-screenshot-longevity-lwt-3h-test-scylla-per-server-metrics-nemesis-20220123_080630-longevity-lwt-3h-master-monitor-node-beaf65ba-1.png |
| 20220123_080431 | grafana        | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_080431/grafana-screenshot-overview-20220123_080431-longevity-lwt-3h-master-monitor-node-beaf65ba-1.png                                                |
| 20220123_082001 | critical       | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/critical-beaf65ba.log.tar.gz                                                                                                                   |
| 20220123_082001 | db-cluster     | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/db-cluster-beaf65ba.tar.gz                                                                                                                     |
| 20220123_082001 | debug          | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/debug-beaf65ba.log.tar.gz                                                                                                                      |
| 20220123_082001 | email_data     | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/email_data-beaf65ba.json.tar.gz                                                                                                                |
| 20220123_082001 | error          | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/error-beaf65ba.log.tar.gz                                                                                                                      |
| 20220123_082001 | event          | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/events-beaf65ba.log.tar.gz                                                                                                                     |
| 20220123_082001 | left_processes | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/left_processes-beaf65ba.log.tar.gz                                                                                                             |
| 20220123_082001 | loader-set     | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/loader-set-beaf65ba.tar.gz                                                                                                                     |
| 20220123_082001 | monitor-set    | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/monitor-set-beaf65ba.tar.gz                                                                                                                    |
| 20220123_082001 | normal         | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/normal-beaf65ba.log.tar.gz                                                                                                                     |
| 20220123_082001 | output         | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/output-beaf65ba.log.tar.gz                                                                                                                     |
| 20220123_082001 | event          | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/raw_events-beaf65ba.log.tar.gz                                                                                                                 |
| 20220123_082001 | sct            | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/sct-beaf65ba.log.tar.gz                                                                                                                        |
| 20220123_082001 | summary        | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/summary-beaf65ba.log.tar.gz                                                                                                                    |
| 20220123_082001 | warning        | https://cloudius-jenkins-test.s3.amazonaws.com/beaf65ba-b35f-4721-bc69-9b17ec0799aa/20220123_082001/warning-beaf65ba.log.tar.gz                                                                                                                    |
+-----------------+----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by slivne** [comment]
maybe the same as #9949

**Comment by bhalevy** [comment]
Decoded backtrace:
```
void seastar::backtrace(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
 (inlined by) seastar::backtrace_buffer::append_backtrace() at ./build/release/seastar/./seastar/src/core/reactor.cc:750
 (inlined by) seastar::print_with_backtrace(seastar::backtrace_buffer&, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:780
seastar::print_with_backtrace(char const*, bool) at ./build/release/seastar/./seastar/src/core/reactor.cc:792
 (inlined by) seastar::sigsegv_action() at ./build/release/seastar/./seastar/src/core/reactor.cc:3663
 (inlined by) operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3649
 (inlined by) __invoke at ./build/release/seastar/./seastar/src/core/reactor.cc:3645
wasmtime_runtime::traphandlers::unix::trap_handler at wasmtime_runtime.9xv9ixev-cgu.15:?
?? ??:0
std::__uniq_ptr_impl >::_M_ptr() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:173
 (inlined by) std::unique_ptr >::get() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:422
 (inlined by) std::unique_ptr >::operator*() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/unique_ptr.h:408
 (inlined by) seastar::foreign_ptr > >::operator*() const at ././seastar/include/seastar/core/sharded.hh:904
 (inlined by) operator() at ./mutation_reader.cc:2110
 (inlined by) seastar::future seastar::futurize >::invoke<(anonymous namespace)::shard_reader::close()::$_32&>((anonymous namespace)::shard_reader::close()::$_32&) at ././seastar/include/seastar/core/future.hh:2135
seastar::futurize::type>::type seastar::smp::submit_to<(anonymous namespace)::shard_reader::close()::$_32>(unsigned int, seastar::smp_submit_to_options, (anonymous namespace)::shard_reader::close()::$_32&&) at ././seastar/include/seastar/core/smp.hh:349
 (inlined by) seastar::futurize::type>::type seastar::smp::submit_to<(anonymous namespace)::shard_reader::close()::$_32>(unsigned int, (anonymous namespace)::shard_reader::close()::$_32&&) at ././seastar/include/seastar/core/smp.hh:376
 (inlined by) (anonymous namespace)::shard_reader::close() at ./mutation_reader.cc:2109
operator() at ./mutation_reader.cc:2685
 (inlined by) seastar::future seastar::futurize >::invoke >&>(multishard_combining_reader::close()::$_49&&, std::unique_ptr<(anonymous namespace)::shard_reader, std::default_delete<(anonymous namespace)::shard_reader> >&) at ././seastar/include/seastar/core/future.hh:2135
 (inlined by) auto seastar::futurize_invoke >&>(multishard_combining_reader::close()::$_49&&, std::unique_ptr<(anonymous namespace)::shard_reader, std::default_delete<(anonymous namespace)::shard_reader> >&) at ././seastar/include/seastar/core/future.hh:2166
 (inlined by) seastar::future seastar::parallel_for_each<__gnu_cxx::__normal_iterator >*, std::vector >, std::allocator > > > >, multishard_combining_reader::close()::$_49>(__gnu_cxx::__normal_iterator >*, std::vector >, std::allocator > > > >, __gnu_cxx::__normal_iterator >*, std::vector >, std::allocator > > > >, multishard_combining_reader::close()::$_49&&) at ././seastar/include/seastar/core/loop.hh:555
 (inlined by) seastar::future seastar::internal::parallel_for_each_impl >, std::allocator > > >&, multishard_combining_reader::close()::$_49>(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&) at ././seastar/include/seastar/core/loop.hh:599
 (inlined by) seastar::future seastar::futurize >::invoke (*&)(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&), std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49>(seastar::future (*&)(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&), std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&) at ././seastar/include/seastar/core/future.hh:2135
 (inlined by) auto seastar::futurize_invoke (*&)(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&), std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49>(seastar::future (*&)(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&), std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&) at ././seastar/include/seastar/core/future.hh:2166
 (inlined by) seastar::future seastar::parallel_for_each >, std::allocator > > >&, multishard_combining_reader::close()::$_49>(std::vector >, std::allocator > > >&, multishard_combining_reader::close()::$_49&&) at ././seastar/include/seastar/core/loop.hh:614
 (inlined by) multishard_combining_reader::close() at ./mutation_reader.cc:2684
flat_mutation_reader::close() at ././flat_mutation_reader.hh:569
 (inlined by) seastar::future<(anonymous namespace)::page_consume_result<(anonymous namespace)::data_query_result_builder> > read_page<(anonymous namespace)::data_query_result_builder>(seastar::shared_ptr, seastar::lw_shared_ptr, query::read_command const&, std::vector, std::allocator > > const&, tracing::trace_state_ptr, (anonymous namespace)::data_query_result_builder&&) at ./multishard_mutation_query.cc:746
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./seastar/src/core/reactor.cc:2344
 (inlined by) seastar::reactor::run_some_tasks() at ./build/release/seastar/./seastar/src/core/reactor.cc:2754
seastar::reactor::do_run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2923
seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2806
seastar::app_template::run_deprecated(int, char**, std::function&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:265
seastar::app_template::run(int, char**, std::function ()>&&) at ./build/release/seastar/./seastar/src/core/app-template.cc:156
scylla_main(int, char**) at ./main.cc:515
std::function::operator()(int, char**) const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/std_function.h:560
 (inlined by) main at ./main.cc:1517
/data/scylla-s3-reloc.cache/by-build-id/d48bc5d3676f53140f58fae39086e98f17a58534/extracted/scylla/libexec/scylladb/libreloc/libc.so.6 0x27b74 
_start at ??:?
```

**Comment by bhalevy** [comment]
Duplicate of #9923

**Comment by bhalevy** [comment]
Fixed by d7a993043de250f39c3881017b83fbb1e789e6fb
